fairseq/training.py

```diff
@@ -0,0 +1,434 @@
+            for k in cnfg_d.keys():
+                if type(cnfg_d[k]) == dict:
+                    get_sis_paths(cnfg_d[k])
+                elif type(cnfg_d[k]) == tk.Path:
+                    cnfg_d[k] = cnfg_d[k].get_path()
+
+        path_corrected_config = self.config_dict.copy()
+        get_sis_paths(path_corrected_config)
+
+        config_yaml = yaml.dump(path_corrected_config)
+        # "# @package _group_" was written at the beginning in the example .yaml from fairseq:
+        if self.yaml_prefix != "":
+            config_yaml = self.yaml_prefix + "\n" + config_yaml
+        with open(path, "w") as file:
+            file.write(config_yaml)
+
+
+class PytorchHydraModel:
+    """
+    Defines a Pytorch hydra model as yaml config, pytorch checkpoint file and epoch
+    """
+
+    def __init__(self, fairseq_hydra_config_file, model, epoch):
+        """
+        :param Path fairseq_hydra_config_file: Path to a returnn config file
+        :param Path model: Path to a pytorch checkpoint
+        :param int epoch:
+        """
+        self.returnn_config_file = fairseq_hydra_config_file
+        self.model = model
+        self.epoch = epoch
+
+
+class FairseqHydraTrainingJob(Job):
+    """
+    Train a Fairseq model using fairseq-hydra-train
+    """
+
+    def __init__(
+        self,
+        fairseq_hydra_config,
+        *,  # args below are keyword only
+        command_line_args=None,
+        max_epoch=None,
+        save_interval=None,
+        keep_epochs=None,
+        time_rqmt=4,
+        mem_rqmt=4,
+        cpu_rqmt=2,
+        gpu_rqmt=1,
+        fairseq_python_exe=None,
+        fairseq_hydra_exe=None,
+        fairseq_root=None,
+        use_cache_manager=True,
+        zipped_audio_dir=None,
+    ):
+        """
+        :param FairseqHydraConfig fairseq_hydra_config:
+        :param list command_line_args: Additional command line arguments (starting with "--*"),
+            to configure the Fairseq-hydra task
+        :param int|None max_epoch: maximum number of epochs to run. Note that this value IS currently HASHED.
+        :param int|None save_interval: save a checkpoint each n-th epoch
+        :param list[int]|set[int]|None keep_epochs: specify which checkpoints are kept in self.out_models.
+            Use None for each save_interval-th epoch
+        :param int|float time_rqmt: Overall time requirements
+        :param int|float mem_rqmt: Memory requirements (per GPU)
+        :param int cpu_rqmt: Required number of CPUs (per GPU)
+        :param int gpu_rqmt: Number of required GPUs
+        :param Path|str python_exe: File path to the executable for running python
+        :param Path|str fairseq_hydra_exe: File path to the python executable for running fairseq-hydra-train.
+            (usually in the same folder as python exe '.../bin/')
+        :param Path|str fairseq_root: File path to the fairseq git for alternative call of fairseq-hydra-train
+            (no need to install fairseq here)
```

Comment:
I think we do not need both ways here, I would prefer specifying the fairseq_root since we don't need to install the fairseq @michelwi which way would you suggest?

Reply:
yes, your suggestion sounds reasonable. This way we can use a CloneGitRepositoryJob.